code/modules/keybindings/_readme.dm

```diff
@@ -0,0 +1,38 @@
+/*
+	This whole system is heavily based off of forum_account's keyboard library. As in I copied and
+	pasted it and then tweaked it to fit our needs. So thanks to forum_account for saving the day.
+
+	.dmf macros have some very serious shortcomings. For example, they do not allow reusing parts
+	of one macro in another, so giving cyborgs their own shortcuts to swap active module couldn't
+	inherit the movement that all mobs should have anyways. The webclient only supports one macro,
+	so having more than one was problematic. Additionally each keybind has to call an actual
+	verb, which meant a lot of hidden verbs that just call one other proc. Also our existing
+	macro was really bad and tied unrelated behavior into Northeast(), Southeast(), Northwest(),
+	and Southwest().
+
+	The basic premise of this system is to not screw with .dmf macro setup at all and handle
+	pressing those keys in the code instead. We have every key call client.keyDown()
+	or client.keyUp() with the pressed key as an argument. Certain keys get processed
+	directly by the client because they should be doable at any time, then we call
+	keyDown() or keyUp() on the client's holder and the client's mob's focus.
```

Comment:
On a more serious note, why is there a delay? Can the keybinding system not handle some events clientside?